### ISSUE TITLE
no-useless-escape false positive

### ISSUE THREAD

**Issue by keirlawson** [eslint/eslint#6514]
**What version of ESLint are you using?**
2.13.1

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
```json
{
  "extends": ["eslint:recommended", "plugin:react/recommended", "plugin:import/errors", "plugin:import/react"],
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "node": true,
    "es6": true,
    "mocha": true
  },
  "plugins": [
    "react",
    "mocha",
    "import"
  ],
  "globals": {
    "PromotionSuite": true,
    "feedback": false,
    "CampaignWizard": false
  },
  "settings": {
    "react": {
      "version": "0.14"
    }
  },
  "rules": {
    "react/wrap-multilines": 2,
    "react/prop-types": 0,

    "mocha/no-exclusive-tests": "error",
    "mocha/no-skipped-tests": 0,
    "mocha/no-pending-tests": "error",
    "mocha/handle-done-callback": "error",

    "eol-last": 2,
    "indent": [ 2, 2, { "SwitchCase": 1 } ],
    "jsx-quotes": [ 2, "prefer-double" ],
    "no-multiple-empty-lines": 2,
    "no-var": 2,
    "quotes": [ 2, "single", {"avoidEscape": true, "allowTemplateLiterals": true} ],
    "semi": [ 2, "never" ],
    "space-before-blocks": [ 2, "always" ],
    "space-before-function-paren": [ 2, { "anonymous": "always", "named": "never" } ],

    "no-warning-comments": 2,
    "eqeqeq": "error",
    "no-param-reassign": "error",
    "no-useless-escape": "error",
    "curly": "error"
  }
}

```

**What did you do? Please include the actual source code causing the issue.**
```js
const re = new RegExp('^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$')
```

**What did you expect to happen?**
No linting error

**What actually happened? Please include the actual, raw output from ESLint.**
`13:53  error  Unnecessary escape character: \  no-useless-escape`

Not prepending an escaping `\` to the `.` changes the meaning of the regex, thus the escape character is not useless.

**Comment by mysticatea** [comment]
Thank you for this issue.

I think this is correct behavior.
You need to escape the `\`.

    const re = new RegExp('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,}$')

**Comment by keirlawson** [comment]
Ah good point, my bad!
